### PR TITLE
Final version delivered for M4

### PR DESCRIPTION
**Pull request by RanoRaraku**
A verified version that contains all deliverables as per the M4 report. Works for both Samromur_general and Samromur_teen.

### CODE REVIEW

File: s5_base/local/samromur_data_prep.sh [hunk 1]
```diff
@@ -15,28 +15,33 @@ set -e
 
 data_type=$2
 data_dir=$3
-audio_src_dir=$1/audio/
-metadata=$1/metadata_v2.tsv
-
+audio_src_dir=$1/audio
```
Comment (judyfong) [inline s5_base/local/samromur_data_prep.sh:18]: Why did you remove the trailing forward slash?

File: s5_base/run.sh [hunk 2]
```diff
@@ -39,250 +39,148 @@ for f in "$lm_train" "$prondict_orig" "$g2p_model"; do \
...
-        nb_oov=$(join -1 1 -2 1 data/$n/vocab_text_only.tmp <(awk '$2 ~ /[[:print:]]/ { print $2" "$1 }' data/$n/words.cnt | sort -k1,1) \
-        | sort | awk '{total = total + $2}END{print total}')
+          <(cut -f1 $prondict | sort -u) > data/$n/vocab_text_only.tmp
+        nb_oov=$(join -1 1 -2 1 data/$n/vocab_text_only.tmp <(awk '$2 ~ /[[:print:]]/ { print $2" "$1 }' \
+           data/$n/words.cnt | sort -k1,1) | sort | awk '{total = total + $2}END{print total}')
         oov=$(echo "scale=3;$nb_oov/$nb_tokens*100" | bc)
-        echo "The out of vocabulary rate for $n is: $oov" >> oov_rate || exit 1;
-    done
+        echo "The out of vocabulary rate for $n is: $oov" || exit 1;
+    done > oov_rate
```
Comment (judyfong) [inline s5_base/run.sh:93]: What exactly does oov_rate do?

File: s5_base/run.sh [hunk 3]
```diff
@@ -39,250 +39,148 @@ for f in "$lm_train" "$prondict_orig" "$g2p_model"; do \
...
-# train a monophone system
+# Subsample train for faster start
 if [ $stage -le 6 ]; then
     echo "Make subsets of the training data to use for the first mono and triphone trainings"
-    
-    utils/subset_data_dir.sh data/train 40000 data/train_40k
-    utils/subset_data_dir.sh --shortest data/train_40k 5000 data/train_5kshort
-    utils/subset_data_dir.sh data/train 10000 data/train_20k
+    utils/subset_data_dir.sh data/train 5000 data/train_5k
+    utils/subset_data_dir.sh data/train 10000 data/train_10k
```
Comment (judyfong) [inline s5_base/run.sh:122]: Why did you reduce the subsets so much? Will it even learn anything with such small subsets?

**Review by judyfong — CHANGES_REQUESTED** [review]
Your commits say you have RESULTS files but I don't see them either on github or locally when i clone this branch so you'll need to re-add that stuff. Also, why are the metadata files labelled r1? And if they're included in the repo, why are they not used in the scripts?

I don't see why we need both a samromur_data_prep in python and bash. I suggest you delete the bash one and integrate the python one with run.sh

I don't understand why you made so many unneeded bash stylistic changes. It makes it harder to read the real changes to the script.

Once again I iterate that we need a standard when commit scripts. Looking at PRs that are a mix of stylistic changes and working changes are a waste of our time. Next time. please do the stylistic changes in their own PR so they won't distract from the work that has been done. So, far we've agreed on 2 spaces per tab for bash and 4 spaces per tab for python. I think we need to also agree on how long a line is. I personally don't like lines longer than 80chars and that's usually a standard line length in formatters. You often have lines around 90chars long.

**Comment by judyfong** [comment]
oh and you should learn to configure your github user.name and user.email

File: s5_base/local/samromur_data_prep.sh (hunk 1 above)
Comment (RanoRaraku) [reply s5_base/local/samromur_data_prep.sh:18]: because when you concatenate it later in the code it, you use / again making // appear. Linux will process this but still an 'error
'

File: s5_base/run.sh (hunk 2 above)
Comment (RanoRaraku) [reply s5_base/run.sh:93]: ask Inga, I think its mainly for info

File: s5_base/run.sh (hunk 3 above)
Comment (RanoRaraku) [reply s5_base/run.sh:122]: I left the 5k and 10k as they were. I removed 40k subset because it will make the script fail on teen data as there is < 40k files originally. This way the script is fully operational on both general and teen.

**Review by judyfong — DISMISSED** [review]

**Comment by RanoRaraku** [comment]
> Your commits say you have RESULTS files but I don't see them either on github or locally when i clone this branch so you'll need to re-add that stuff. Also, why are the metadata files labelled r1? And if they're included in the repo, why are they not used in the scripts?

RESULTS have been added back. They are labelled as r1 to distinguish from whatever later versions will occur as we keep adding data. So for future to be able to tell difference. I added them because Anna wanted to see them, but disassociating metadata from actual audio would mean I would have to rewrite the data prep script. I see no reason to since after the dataset are public everything will work.

> I don't see why we need both a samromur_data_prep in python and bash. I suggest you delete the bash one and integrate the python one with run.sh

I see neither but lets keep it. I prefer bash but lets make a decision on which one when we deliver the final product.

> I don't understand why you made so many unneeded bash stylistic changes. It makes it harder to read the real changes to the script.

Because reading one command per line code made my eyes hurt and my screen too small for even a single compound command.

> Once again I iterate that we need a standard when commit scripts. Looking at PRs that are a mix of stylistic changes and working changes are a waste of our time. Next time. please do the stylistic changes in their own PR so they won't distract from the work that has been done. So, far we've agreed on 2 spaces per tab for bash and 4 spaces per tab for python. I think we need to also agree on how long a line is. I personally don't like lines longer than 80chars and that's usually a standard line length in formatters. You often have lines around 90chars long.

I applaud the initiative. I use 80-100 chars a line standard but I am willing to compromise.

**Review by judyfong — APPROVED** [review]

File: s5_base/run.sh (hunk 2 above)
Comment (ingarun) [reply s5_base/run.sh:93]: It is just information. If we have high oov_rate in the test sets then that explains partly why we have a high WER

**Review by ingarun — COMMENTED** [review]
I think it is fine. But I find it very difficult to see what is happening in the commands. i find it a lot easier to read them one per line

**Comment by RanoRaraku** [comment]
> I think it is fine. But I find it very difficult to see what is happening in the commands. i find it a lot easier to read them one per line

I see. I have a complete opposite view. I dont think there is a solution that would make us both happy then.
